malloc/malloc.c

```diff
@@ -10,9 +11,18 @@
     You may want to use the string_length function to figure out the
     length of the input string.
 */
-char *string_dup(char *src)
-{
+char *string_dup(char *src) {
+    char *dup = NULL;
+    int length = strlen(src);
+    char *s = malloc(length);
+    dup = s;
+
+    for ( int i = 0; i < length; i++) {
+        *s = src[i];
```

Comment:
Instead of having the `dup` variable and incrementing `s` each time, you could just do `s[i] = src[i]`